core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -40,6 +46,15 @@
     static final FilteredQueryBuilder PROTOTYPE = new FilteredQueryBuilder(null, null);
 
     /**
+     * A query that applies a filter to the results of a match_all query.
+     * @param filterBuilder The filter to apply on the query (Can be null)
+     * */
+    public FilteredQueryBuilder(@Nullable QueryBuilder filterBuilder) {
+        this.queryBuilder = new MatchAllQueryBuilder();
```

Comment:
shall we use MatchAllQueryBuilder as a default value in general, outside of this constructor? If not set, it will be match all? When set, we check that it's not set to null in setter.